collectors/ciscomeraki/collector.js

```diff
@@ -0,0 +1,270 @@
+                    collector.reportApiThrottling(function () {
+                        return callback(null, [], state, state.poll_interval_sec);
+                    });
+                }
+                // set errorCode if not available in error object to showcase client error on DDMetric
+                else {
+                    if (error && error.response && error.response.status == 404) {
+                        console.error('error.response.status>>>', error.response.status, JSON.stringify(state));
+                        state.retry = state.retry ? state.retry + 1 : 1;
+                        if (state.retry > 3) {
+                            AlLogger.debug(`CMRI0000021 Deleted SQS message from Queue${JSON.stringify(state)}`);
+                            collector._invokeContext.succeed();
+                        } else {
+                            return callback(error);
+                        }
+                    }
+                    else if (error && error.response && error.response.data) {
+                        AlLogger.debug(`CMRI0000022 error ${error.response.data.errors} - status: ${error.response.status}`);
+                        error.response.data.errorCode = error.response.status;
+                        return callback(error.response.data);
+                    }
+                    else {
+                        AlLogger.debug(`CMRI0000023 errors ${error.response.data.errors} - status: ${error.response.status}`);
+                        return callback(error);
+                    }
+                }
+            });
+    }
+
+    validateAndPrepare(callback) {
+        let collector = this;
+        const clientSecret = collector.secret;
+        if (!clientSecret) {
+            return callback("The Client Secret was not found!");
+        }
+
+        const apiEndpoint = process.env.paws_endpoint.replace(/^https:\/\/|\/$/g, '');
+        const orgKey = process.env.paws_collector_param_string_2;
+        if (!orgKey) {
+            return callback("orgKey was not found!");
+        }
+
+        return { clientSecret, apiEndpoint, orgKey };
+    }
+    _getNextCollectionState(curState) {
+        const untilMoment = moment(curState.until);
+        const { nextUntilMoment, nextSinceMoment, nextPollInterval } = calcNextCollectionInterval('no-cap', untilMoment, this.pollInterval);
+        return {
+            networkId: curState.networkId,
+            since: nextSinceMoment.toISOString(),
+            until: nextUntilMoment.toISOString(),
+            nextPage: null,
+            poll_interval_sec: nextPollInterval
+        };
+    }
+
+    _getNextCollectionStateWithNextPage({ networkId, since, until }, nextPage) {
+        const obj = {
+            networkId,
+            since: nextPage,
+            until,
+            nextPage: null,
+            poll_interval_sec: 1
+        };
+        return obj;
+    }
+
+    pawsGetRegisterParameters(event, callback) {
```

Comment:
Looks like not used anywhere’s .Plz double check.

Reply:
this is generated by default when new collector is created